src/_pytest/python.py

```diff
@@ -733,21 +736,16 @@ def _collectfile(
         return ihook.pytest_collect_file(file_path=fspath, parent=self)  # type: ignore[no-any-return]
 
     def collect(self) -> Iterable[Union[nodes.Item, nodes.Collector]]:
-        this_path = self.path.parent
-
         # Always collect the __init__ first.
-        if self.session.isinitpath(self.path) or path_matches_patterns(
```

Comment:
Since `__init__.py` is no longer treated special by `pytest_pycollect_makemodule`, can just use `_collectfile` for it.